src/tools/crawler/network.rs

```diff
@@ -50,6 +50,16 @@ impl KnownNetwork {
         node.connecting_time = Some(connecting_time);
         node.server = Some(server_version);
     }
+
+    /// Increases connection failures to the `addr` and returns its new value.
+    pub(super) async fn increase_connection_failures(&self, addr: SocketAddr) -> u8 {
+        let mut nodes = self.nodes.write().await;
+        let mut node = nodes.get_mut(&addr).unwrap();
+        if node.connection_failures < u8::MAX {
```

Comment:
Instead of whole `if` block we could use `saturating_add` of u8 type (https://doc.rust-lang.org/std/primitive.u8.html#method.saturating_add).

Reply:
Done.